docs_src/general/EdgeX_CN.md

```diff
@@ -0,0 +1,153 @@
+
+### CN Ingredients Not In EdgeX (and why)
+
+**Kubernetes**
+
+EdgeX provides example Helm Charts to assist adopters that want to run EdgeX in a Kubernetes environment.  However, EdgeX was not designed to fully operate in a multi-cluster environment and take advantage of a full K8s environment.  Our example Helm Charts, for example, allow a single instance of each EdgeX service to be deployed/orchestrated and monitored, but it would not allow K8s to fully manage and scale EdgeX services.  Why?  First and foremost Kubernetes is large compared to the resource constraints of some edge platforms.  While smaller Kubernetes environments are being developed for the edge (see *Futures* below), a whole host of challenges such as resource constraints, environment, infrastructure, etc. (as mentioned under *Containers* above) may not allow K8s to operate at the edge.  Kubernetes is, for the most part, about the ability to load balance, distribute traffic, and scale (up or down) workloads so that an application remains stable.  But on an edge platform, where would Kubernetes find the resources to balance and distribute and scale?  Because edge nodes are static and often times physically connected to the sensors they collect data from, there is not the means to grow and/or shift the workloads.  Portions of EdgeX might be able to scale up or down (those not physically tied to an edge sensor), but the platform as a whole is often rooted to the physical world it is connected to.
```

Comment:
This paragraph spends a lot of time denigrating Kubernetes and doesn't acknowledge why people would *want* to use Kubernetes at the edge:

- It provides a "central pane of glass" for placing workloads at the edge, monitoring them, and being able to easily upgrade them, more easily than a native, snap-based, or Docker-based deployment.
- It allows people to more easily deploy workloads that span from the cloud to the edge by using familiar tools that allow users to place their workloads in a more appropriate place.
- The world has clearly chosen Kubernetes over Docker for container orchestration, with lots of commercially supported Kubernetes distributions for doing so.
- Despite the fact that edge resources are not elastic, Kubernetes is possible of making better scheduling decisions in a complex edge environment, where computational accelerators may be available on some nodes and not others, and Kubernetes can help place those workloads where they will run most efficiently.

That said, there are a number of caveats:
- As mentioned, edge resources are not elastic
- As explained earlier, some devices are physically connected to nodes using non-routable or non-Internet protocols, which reduces the value of the Kubernetes scheduler
- Storage is a sticking point - unless there is enough infrastructure at the edge to make storage highly available, separation of the storage from the workload mathematically reduces availability (i.e: 0.9 x 0.9 = 0.81 !)
- Available network bandwidth and latency can be a concern: a Kubernetes cluster generates a lot of background network and CPU activity.

Reply:
@bnevis-i - this is some great commentary.  I didn't mean to denigrate K8s (and I don't think I really did) but I certainly didn't highlight the pros/cons appropriate.  I have added your lists to the discussion in this area.  See if it works to help address this concern.